### ISSUE TITLE
[Question]Whether LightGBM uses 'Goss' in GBDT by default?

### ISSUE THREAD

**Issue by Lucienxhh** [microsoft/LightGBM#4284]
It seems contradictory that:

1.In the python package of lightgbm, the default option of parameter 'boosting' is 'gbdt' while the optional option is 'goss';

2.The paper of lightgbm proposes Goss based on the gradient for each data instance in GBDT.

I'm very confused whether lightgbm uses 'goss' in gbdt by default? if it does, how does it work for every feature? it works before 'histogram' or during generation of tree ?

**Comment by shiyu1994** [comment]
@Lucienxhh Thanks for using LightGBM. LightGBM uses `gbdt` as `boosting_type` by default, instead of `goss`. Note that `goss` still uses the histogram method as `gbdt` does, the only difference is which data are sampled. With `gbdt`, the whole training set is used, while with `goss`, the dataset is sampled as the paper describes. The tree training method is the same for `goss` and `gbdt`.

**Comment by StrikerRUS** [comment]
@Lucienxhh 
Also, these discussions may help better understand GOSS: #3348 and https://github.com/microsoft/LightGBM/issues/4086#issuecomment-805548930.

**Comment by Lucienxhh** [comment]
> @Lucienxhh Thanks for using LightGBM. LightGBM uses `gbdt` as `boosting_type` by default, instead of `goss`. Note that `goss` still uses the histogram method as `gbdt` does, the only difference is which data are sampled. With `gbdt`, the whole training set is used, while with `goss`, the dataset is sampled as the paper describes. The tree training method is the same for `goss` and `gbdt`.

@shiyu1994 Thanks for your reply. According to your description, 'goss' is a method to reduce the number of samples while 'gbdt' is a method to build tree. So why 'goss' is one of the boosting types ?  It 's very confusing for a new hand like me.

**Comment by shiyu1994** [comment]
@Lucienxhh In `goss`, the gradients and hessians are reweighted, in addition to sampling. I think it makes sense to make it a type of boosting, since it affects the calculation of gradients and hessians.
